### PR TITLE
feat: button-group component

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | Originated from https://github.com/ripe-tech/ripe-twitch/issues/7 (payment buttons) |
| Dependencies | -- |
| Decisions | Similar to ripe-white [`button-group`](https://github.com/ripe-tech/ripe-white/blob/4fa823a3fc0658eacbfaa3247caee575f2c283c6/vue/components/molecules/button-group/button-group.vue). Only added an additional slot to possibly modify the interior of the button, as well as passing the color prop to the `button-color`.  |
| Link | https://ripe-components-vue-git-ms-7-buttongroupcomponent.platforme.vercel.app/?path=/story/molecules--button-group | 
| Animated GIF | button-group |

### CODE REVIEW

File: vue/components/ui/molecules/button-group/button-group.vue [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            this.$emit("update:value", value);
+        }
+    },
+    methods: {
+        buttonClasses(item) {
+            const base = {
+                selected: this.valueData === item.value
+            };
+            base[item.value] = true;
+            return base;
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-group/button-group.vue:105]: may improve readability like this
```suggestion
            const base = {};
            base[item.value] = true;
            base.selected = this.valueData === item.value;
            return base;
```

File: vue/components/ui/molecules/button-group/button-group.stories.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        template: `
+            <button-group
+                v-bind:items="items"
+                v-bind:disabled="disabled"
+            >
+                <template v-slot:button-3>
+                    Special Item 3
+                </template>
+            </button-group>
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-group/button-group.stories.js:31]: can be enriched with a `<div> {{ valueData }} </div>` with the currently selected value and use `v-bind:value.sync="valueData"`

File: vue/components/ui/molecules/button-group/button-group.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            class="button-container"
+            v-for="item in items"
+            v-bind:key="item.value"
+            v-on:click="onClick($event, item)"
+        >
+            <slot v-bind:item="item" v-bind:selected="valueData === item.value">
+                <button-color
+                    class="button-color"
+                    v-bind:class="buttonClasses(item)"
+                    v-bind:disabled="disabled"
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-group/button-group.vue:13]: could be useful if we we're able disable individual buttons 
```suggestion
                    v-bind:disabled="disabled || item.disabled"
```

File: vue/components/ui/molecules/button-group/button-group.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    methods: {
+        buttonClasses(item) {
+            const base = {
+                selected: this.valueData === item.value
+            };
+            base[item.value] = true;
+            return base;
+        },
+        onClick(event, item) {
+            if (this.disabled) return;
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-group/button-group.vue:108]:
```suggestion
            if (this.disabled || item.disabled) return;

```

**Review by 3rdvision — CHANGES_REQUESTED** [review]

**Review by 3rdvision — APPROVED** [review]

**Review by gcandal — APPROVED** [review]

**Review by gcandal — CHANGES_REQUESTED** [review]
Move this to ripe-twitch

**Comment by joamag** [comment]
I'm closing this PR here, let's migrate this to Twitch UI instead.

**Pull request closed without merging**